packages/snjs/lib/services/settings_service/SettingsRepo.ts

```diff
@@ -0,0 +1,100 @@
+import { SettingName, SettingPayload } from '@standardnotes/settings';
+import * as messages from '../api/messages';
+import { SettingsProvider } from './SettingsProvider';
+import {
+  DeleteSettingResponse,
+  GetSettingResponse,
+  ListSettingsResponse,
+  UpdateSettingResponse,
+  User,
+} from '../api/responses';
+import { UuidString } from '@Lib/types';
+
+interface SettingsAPI {
+  listSettings(userUuid: UuidString): Promise<ListSettingsResponse>;
+
+  updateSetting(
+    userUuid: UuidString,
+    settingName: string,
+    settingValue: string
+  ): Promise<UpdateSettingResponse>;
+
+  getSetting(
+    userUuid: UuidString,
+    settingName: string
+  ): Promise<GetSettingResponse>;
+
+  deleteSetting(
+    userUuid: UuidString,
+    settingName: string
+  ): Promise<DeleteSettingResponse>;
+}
+
+export class SettingsRepo implements SettingsProvider {
+  constructor(
+    private readonly api: SettingsAPI,
+    private readonly getUser: () => User | null
+  ) {}
+
+  canModify() {
```

Comment:
Can this be more descriptive? Perhaps

```suggestion
  isReady() {
```

or 

```suggestion
  isReadyForModification() {
```

Reply:
Done. Resolving comment.